### ISSUE TITLE
Ruby layer quote and operator highlighting

### ISSUE THREAD

**Issue by robmerrell** [syl20bnr/spacemacs#1332]
When setting a dark colored theme as the startup theme the quotes and operators in Ruby files are hard to see because they are black. I tracked it down to this section of code in the Ruby layer, but don't know enough elisp to debug why it is happening: https://github.com/syl20bnr/spacemacs/blob/master/contrib/lang/ruby/packages.el#L43

**Comment by x-ji** [comment]
Happens for me on Zenburn theme also. Though have not seen it on other dark themes actaully.

**Comment by icambron** [comment]
Happens to me in Wheatgrass. I did find *some* dark themes where it works OK, though.

**Comment by danielberkompas** [comment]
I'm seeing this on the `ujelly` theme.

**Comment by mosic** [comment]
I have the same problem using `seti` theme.

Replacing

``` lisp
(defun erm-darken-color (name)
  (let ((attr (face-attribute name :foreground)))
    (unless (equal attr 'unspecified)
      (color-darken-name attr 20) "#000000"))))))
```

with 

``` lisp
(defun erm-darken-color (name)
  (color-darken-name (face-attribute name :foreground) 20)))))
```

seems to fix the issue. I would love a proper solution though.

**Comment by icambron** [comment]
@mosic I'm a little curious about that as a fix - it seems like the code was trying to protect us from calling `color-darken-name` on an unspecified attribute, presumably because that breaks? And if we're seeing that black everywhere, then I'd have guessed it really *was* unspecified, which is why it surprises me that your fix doesn't cause the breakage the original code was there to prevent...

Perhaps the `color-darken-name` was fixed upstream to not barf on bad face attributes?

**Comment by mosic** [comment]
@icambron You're right, this is the same as not overriding the `erm-darken-color` function. Not sure why I did it this way, it was a couple of weeks ago (probably for no good reason).

I too assumed it was fixed upstream in the meantime but that doesn't seem to be the case: https://github.com/zenspider/enhanced-ruby-mode/commits/master. Maybe I'm missing something.

Anyway just reverting the first block of this commit: https://github.com/syl20bnr/spacemacs/commit/30ced0a08bbc672331244a3bcb343c364ed13611 / removing the override seems to work in the couple of themes I tried, not sure if there's a regression somewhere else.
